nixos/lib/test-driver/test-driver.py

```diff
@@ -812,7 +813,8 @@ def process_serial_output() -> None:
                 self.last_lines.put(line)
                 self.log_serial(line)
 
-        _thread.start_new_thread(process_serial_output, ())
+        self.serial_thread = threading.Thread(target=process_serial_output)
+        self.serial_thread.start()
```

Comment:
I think we are just missing the `serial_thread.join()` at the end from https://github.com/NixOS/nixpkgs/commit/a874235dff32bd77125034cfd9542a91b68efb03